diff --git a/linker/linker.cpp b/linker/linker.cpp
index fd1927876..6300d437e 100644
--- a/linker/linker.cpp
+++ b/linker/linker.cpp
@@ -1612,8 +1612,8 @@ static int open_library_in_zipfile(ZipArchiveCache* zip_archive_cache,
   if (realpath_fd(fd, realpath)) {
     *realpath += separator;
   } else {
-    PRINT("warning: unable to get realpath for the library \"%s\". Will use given path.",
-          normalized_path.c_str());
+    //PRINT("warning: unable to get realpath for the library \"%s\". Will use given path.",
+    //      normalized_path.c_str());
     *realpath = normalized_path;
   }
 
@@ -1657,7 +1657,7 @@ static bool open_load_library_at_path(ZipArchiveCache* zip_archive_cache,
     if (fd != -1) {
       file_offset = 0;
       if (!realpath_fd(fd, &realpath)) {
-        PRINT("warning: unable to get realpath for the library \"%s\". Will use given path.", path);
+        //PRINT("warning: unable to get realpath for the library \"%s\". Will use given path.", path);
         realpath = path;
       }
     }
@@ -1924,8 +1924,8 @@ static bool load_library(android_namespace_t* ns,
     }
 
     if (!realpath_fd(extinfo->library_fd, &realpath)) {
-      PRINT("warning: unable to get realpath for the library \"%s\" by extinfo->library_fd. "
-            "Will use given name.", name);
+      //PRINT("warning: unable to get realpath for the library \"%s\" by extinfo->library_fd. "
+      //      "Will use given name.", name);
       realpath = name;
     }
 
